driver-scala/src/main/scala/org/mongodb/scala/ObservableImplicits.scala

```diff
@@ -85,89 +84,49 @@ trait ObservableImplicits {
      */
     def toSingle(): SingleObservable[T] = this
 
-    override def subscribe(observer: Observer[_ >: T]): Unit = {
-      publisher.subscribe(
-        SubscriptionCheckingObserver(new Observer[T]() {
-          @volatile
-          var results: Option[T] = None
-
-          @volatile
-          var terminated: Boolean = false
-
-          override def onSubscribe(subscription: Subscription): Unit = {
-            observer.onSubscribe(subscription)
-            subscription.request(1)
-          }
-
-          override def onError(throwable: Throwable): Unit = completeWith("onError", () => observer.onError(throwable))
-
-          override def onComplete(): Unit = {
-            completeWith("onComplete", { () =>
-              results.foreach { (result: T) =>
-                observer.onNext(result)
-              }
-              observer.onComplete()
-            })
-          }
-
-          override def onNext(tResult: T): Unit = {
-            check(results.isEmpty, "SingleObservable.onNext cannot be called with multiple results.")
-            results = Some(tResult)
-          }
-
-          private def completeWith(method: String, action: () => Any): Unit = {
-            check(!terminated, s"$method called after the Observer has already completed or errored. $observer")
-            terminated = true
-            action()
-          }
-
-          private def check(requirement: Boolean, message: String): Unit = {
-            if (!requirement) throw new IllegalStateException(message)
-          }
-        })
-      )
-    }
+    override def subscribe(observer: Observer[_ >: T]): Unit = Mono.from(publisher).subscribe(observer)
   }
 
   implicit class ToSingleObservableInt(pub: => Publisher[java.lang.Integer]) extends SingleObservable[Int] {
     val publisher = pub
-    override def subscribe(observer: Observer[_ >: Int]): Unit =
-      publisher.toObservable().map(_.intValue()).toSingle().subscribe(observer)
+    override def subscribe(observer: Observer[_ >: Int]): Unit = {
+      Mono
+        .from(MapObservable(publisher.toObservable(), (i: Integer) => i.toInt))
```

Comment:
Can't use `Mono#map` due to troubles in scala 2.11 and 2.12 expressing `Function<? super T, ? extends V> mapper`.